alerting/alerter-war/src/main/java/org/hawkular/metrics/alerter/ConditionManager.java

```diff
@@ -0,0 +1,554 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.alerter;
+
+import static org.hawkular.metrics.model.MetricType.AVAILABILITY;
+import static org.hawkular.metrics.model.MetricType.COUNTER_RATE;
+import static org.hawkular.metrics.model.MetricType.GAUGE_RATE;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.annotation.Resource;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+
+import org.hawkular.alerts.api.model.condition.Condition;
+import org.hawkular.alerts.api.model.condition.ExternalCondition;
+import org.hawkular.alerts.api.model.data.Data;
+import org.hawkular.alerts.api.model.event.Event;
+import org.hawkular.alerts.api.model.trigger.Trigger;
+import org.hawkular.alerts.api.services.AlertsService;
+import org.hawkular.alerts.api.services.DefinitionsEvent;
+import org.hawkular.alerts.api.services.DefinitionsEvent.Type;
+import org.hawkular.alerts.api.services.DefinitionsListener;
+import org.hawkular.alerts.api.services.DefinitionsService;
+import org.hawkular.metrics.alerter.ConditionExpression.EvalType;
+import org.hawkular.metrics.alerter.ConditionExpression.Query;
+import org.hawkular.metrics.core.service.MetricsService;
+import org.hawkular.metrics.model.BucketPoint;
+import org.hawkular.metrics.model.Buckets;
+import org.hawkular.metrics.model.Metric;
+import org.hawkular.metrics.model.MetricId;
+import org.hawkular.metrics.model.MetricType;
+import org.hawkular.metrics.model.Percentile;
+import org.hawkular.metrics.model.exception.RuntimeApiError;
+import org.hawkular.metrics.model.param.Tags;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.notifications.Listener;
+import org.infinispan.notifications.cachemanagerlistener.annotation.ViewChanged;
+import org.infinispan.notifications.cachemanagerlistener.event.ViewChangedEvent;
+import org.jboss.logging.Logger;
+
+import rx.Observable;
+
+/**
+ * Manages the Metrics expression evaluations and interacts with the Alerts system.
+ *
+ * @author Jay Shaughnessy
+ * @author Lucas Ponce
+ */
+@Startup
+@Singleton
+public class ConditionManager {
+    private final Logger log = Logger.getLogger(ConditionManager.class);
+
+    private static final String TAG_EXTERNAL_CONDITION_NAME = "HawkularMetrics";
+    private static final String TAG_EXTERNAL_CONDITION_VALUE = "MetricsCondition";
+
+    private static final Integer THREAD_POOL_SIZE = 20;
+
+    ScheduledThreadPoolExecutor expressionExecutor;
```

Comment:
Note to self.
ThreadPools are growing.
I guess that at some point it could be nice to have a wrapper for these classes and put more descriptive names to them.
Then when we dump a JVM we can trace better the threads involved.

Just a side comment, no action required, but it is not first time that I think on it.